service/history/workflow/fx.go

```diff
@@ -32,6 +32,7 @@ import (
 
 var Module = fx.Options(
 	fx.Provide(NewCacheFnProvider),
+	fx.Populate(&taskGeneratorProvider),
```

Comment:
hmmm, mixed feeling here... How bad is it to pass thought this along the dependency chain (get it from constructor instead of global var)?

Reply:
I do have the version that pass it along the dependency chain locally. That requires updating (adding the provider to those components) workflow cache, workflow context, mutable state, and everything creates a workflow context or mutable state (e.g. ndc conflict resolver, state rebuilder, resetter, etc...). In total that requires changes to ~40 files.

Retrieving the provider from shard context is also not an option, as it will lead to cycle dependency issue.

Reply:
Is this the only thing that needs to be passed around? Or we already have some things that are passed around everywhere?

Reply:
Most components that need to be passed around are retrieved from shard context (which is probably not a good practice).
 
Workflow cache is passed around directly (only to queue processor related components). Can't get it from shard due to cycle dependency issue. Other stateless utility components are created based on needs (e.g. workflow resetter, delete manager, etc.)

So for now this is the only one besides workflow cache. If in the future we need to have more, then we should probably just inject the entire shard context at that time.